src/test/java/eu/europeana/metis/sandbox/service/dataset/DatasetLogServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+
+  public static final String MESSAGE_2 = "ExceptionMessage2";
+
+  public static final String MESSAGE_3 = "ExceptionMessage3";
+  @Mock
+  private DatasetRepository datasetRepository;
+  @Mock
+  private DatasetLogRepository datasetLogRepository;
+  @InjectMocks
+  private DatasetLogServiceImpl service;
+  @Mock
+  private DatasetEntity dataset;
+  @Captor
+  private ArgumentCaptor<DatasetLogEntity> logCaptor;
+
+  @BeforeEach
+  public void setup() {
+    when(datasetRepository.findById(1)).thenReturn(Optional.of(dataset));
+  }
+
+  @Test
+  public void logException_shouldLogMessageFromSingleException() {
+    Throwable e = new RuntimeException(MESSAGE_1);
+    
+    service.logException(DATASET_ID, e);
+
+    verify(datasetLogRepository).save(logCaptor.capture());
+    DatasetLogEntity savedLog = logCaptor.getValue();
+    assertThat(savedLog.getMessage()).contains(MESSAGE_1);
+    assertThat(savedLog.getStackTrace()).isNotBlank();
+    assertThat(savedLog.getStatus()).isEqualTo(Status.FAIL);
+    assertThat(savedLog.getDataset()).isEqualTo(dataset);
+  }
+
+  @Test
+  public void logException_shouldLogExceptionWithNullMessage() {
+    Throwable e = new RuntimeException();
+
+    service.logException(DATASET_ID, e);
+
+    verify(datasetLogRepository).save(logCaptor.capture());
+    DatasetLogEntity savedLog = logCaptor.getValue();
+    assertThat(savedLog.getMessage()).contains(RuntimeException.class.getName());
+    assertThat(savedLog.getStackTrace()).isNotBlank();
+    assertThat(savedLog.getStatus()).isEqualTo(Status.FAIL);
+    assertThat(savedLog.getDataset()).isEqualTo(dataset);
+  }
+
+  @Test
+  public void logException_shouldLogMessageFromExceptionChain() {
+    Throwable e = new RuntimeException(MESSAGE_3,new Exception(MESSAGE_2, new RuntimeException(MESSAGE_1)));
+
+    service.logException(DATASET_ID, e);
+
+    verify(datasetLogRepository).save(logCaptor.capture());
+    DatasetLogEntity savedLog = logCaptor.getValue();
+    assertThat(savedLog.getMessage()).contains(MESSAGE_1);
+    assertThat(savedLog.getMessage()).contains(MESSAGE_2);
+    assertThat(savedLog.getMessage()).contains(MESSAGE_3);
+    assertThat(savedLog.getStackTrace()).isNotBlank();
+    assertThat(savedLog.getStatus()).isEqualTo(Status.FAIL);
+    assertThat(savedLog.getDataset()).isEqualTo(dataset);
+  }
+
+  @Test
+  public void logException_shouldUnwrapComletionException() {
+    Throwable e = new CompletionException(MESSAGE_2,new RuntimeException(MESSAGE_1));
+
+    service.logException(DATASET_ID, e);
+
+    verify(datasetLogRepository).save(logCaptor.capture());
+    DatasetLogEntity savedLog = logCaptor.getValue();
+    assertThat(savedLog.getMessage()).contains(MESSAGE_1);
+    assertThat(savedLog.getMessage()).doesNotContain(MESSAGE_2);
+    assertThat(savedLog.getStackTrace()).isNotBlank();
+    assertThat(savedLog.getStatus()).isEqualTo(Status.FAIL);
+    assertThat(savedLog.getDataset()).isEqualTo(dataset);
+  }
+
```

Comment:
Perhaps another unit test here? 
To check that there are zero logs when dataset processing goes well

Reply:
To test the case that no logs are saved in case of successful dataset execution, I added tests to the DatasetControllerTest.